tools/devtool

```diff
@@ -1637,6 +1653,73 @@ cmd_install() {
     install -m 755 "$CARGO_SECCOMPILER_TARGET_DIR/$target/$profile/seccompiler-bin" "$install_path"
 }
 
+# Build a Firecracker CI compatible kernel image.
+# Example: `./tools/devtool build_rootfs -c resources/microvm-kernel-arm64.config`
+#
+cmd_build_kernel() {
+    nprocs=$(getconf _NPROCESSORS_ONLN)
+
+    # Parse any command line args.
+    while [ $# -gt 0 ]; do
+        case "$1" in
+            "-h"|"--help")      { cmd_help; exit 1; } ;;
+            "-c"|"--config")
+                shift
+                local KERNEL_CFG="$1"
+                ;;
+            "-n"|"--nproc")
+                shift
+                nprocs="$1"
+                ;;
+            *)
+                die "Unknown argument: $1. Please use --help for help."
+            ;;
+        esac
+        shift
+    done
+
+    # It is mandatory to provide a valid config file for building the kernel.
+    [ -z "$KERNEL_CFG" ] && die "You need to provide the path to a valid kernel config!"
+
+    arch=$(uname -m)
+    if [ "$arch" = "x86_64" ]; then
+        target="vmlinux"
+        cfg_pattern="x86"
+        format="elf"
+    elif [ "$arch" = "aarch64" ]; then
+        target="Image"
+        cfg_pattern="arm64"
+        format="pe"
+    else
+        die "Unsupported architecture!"
+    fi
+
+    kernel_dir_host="$FC_BUILD_DIR"/kernel
+    kernel_dir_ctr="$CTR_FC_BUILD_DIR"/kernel
+    create_dir "$kernel_dir_host"
+
+    # Extract the kernel version from the config file provided as parameter.
+    KERNEL_VERSION=$(cat "$KERNEL_CFG" | grep -Po "^# Linux\/$cfg_pattern (([0-9]+.){2}[0-9]+)" | cut -d ' ' -f 3)
+    validate_kernel_version "$KERNEL_VERSION"
+
+    rust_vmm_recipe_url="https://raw.githubusercontent.com/rust-vmm/vmm-reference/fc_down/resources/kernel/make_kernel.sh"
```

Comment:
Do we have any mechanisms in place to make sure that we can still run this file if the rust-vmm upstream changes?

What about fixing it to a commit ID?

Reply:
Fixing it to a commit ID sounds good to me. @serban300 what are your thoughts on this solution?

Reply:
Sorry, but personally in this particular case I still don't see a great value in reusing the code from rust-vmm and I would still just copy it since there are just a few lines from what I can tell:

```
get linux sources (with wget or clone)
copy the config
yes "" | make oldconfig
make -j [num_configed_threads] vmlinux (or make Image if we're on an aarch64 platform)
```

Also being so simple I don't see any fixes, improvements, or features that will make this script more appealing in the future.

It's true that fixing the script to a particular commit is more reliable, but it still has downsides. For example that commit can be deleted even if it's very unlikely.

Reply:
We are not deleting commits in rust-vmm, that is for sure :D.

Consuming it from the rust-vmm/vmm-reference crate is not the best outcome for the long term, but it looks like a good enough solution in the short term until we work towards the higher goal of having reusable scripts for building resources that can be reused by the multiple projects.

Reply:
Cool, good to know commits are not going to be deleted. thanks @andreeaflorescu . Indeed, vmm-reference is not bound to remain the crate where these test resources are going to be kept. I do think that for the long term we it is best to put these in a separate dedicated repo. However, consuming this code from rust-vmm will ensure we will reach the best version of this script, one that will hopefully be consumed by as many rust powered vms out there. 

@firecracker-microvm/compute-capsule can I get your opinion on this too?

Reply:
@gc-plp I fixed the script to a hardcoded commit, PTAL!

Reply:
Anyway this is not a blocker for me and otherwise the PR looks good.